File: common-module/src/main/java/ru/kraser/technical_helper/common_module/dto/breakage/BreakageFullDto.java
package ru.kraser.technical_helper.common_module.dto.breakage;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Builder;
import ru.kraser.technical_helper.common_module.dto.breakage_comment.BreakageCommentFrontDto;
import ru.kraser.technical_helper.common_module.enums.Priority;
import ru.kraser.technical_helper.common_module.enums.Status;

import java.time.LocalDateTime;
import java.util.List;

@Builder
public record BreakageFullDto(
        String id,
        String departmentId,
        String breakageExecutorId,
        String departmentName,
        String room,
        String breakageTopic,
        String breakageText,
        Status status,
        Priority priority,
        String breakageExecutor,
        String executorAppointedBy,
        String createdBy,
        @JsonFormat(pattern = "MM-dd-yyyy HH:mm:ss")
        LocalDateTime createdDate,
        String lastUpdatedBy,
        @JsonFormat(pattern = "MM-dd-yyyy HH:mm:ss")
        LocalDateTime lastUpdatedDate,
        @JsonFormat(pattern = "MM-dd-yyyy HH:mm:ss")
        LocalDateTime deadline,
        List<BreakageCommentFrontDto> comments
) {
}

File: frontend/src/app/model/breakage/breakage-full-dto.ts
import { Priority } from "../../enum/priority.enum";
import { Status } from "../../enum/status.enum";

export interface BreakageFullDto {
    id: String;
    departmentName: String;
    room: String;
    breakageTopic: String;
    breakageText: String;
    status: Status;
    priority: Priority;
    executor: String;
    executorAppointedBy: String;
    createdBy: String;
    createdDate: Date;
    lastUpdatedBy: String;
    lastUpdatedDate: Date;
    deadline: Date;
    comments: Array<any>;
}